pkg/http/api.go

```diff
@@ -1091,13 +1091,24 @@ func (a *api) onPublish(reqCtx *fasthttp.RequestCtx) {
 
 	body := reqCtx.PostBody()
 	contentType := string(reqCtx.Request.Header.Peek("Content-Type"))
+	metadata := getMetadataFromRequest(reqCtx)
 
 	// Extract trace context from context.
 	span := diag_utils.SpanFromContext(reqCtx)
 	// Populate W3C traceparent to cloudevent envelope
 	corID := diag.SpanContextToW3CString(span.SpanContext())
 	envelope := pubsub.NewCloudEventsEnvelope(uuid.New().String(), a.id, pubsub.DefaultCloudEventType, corID, topic, pubsubName, contentType, body)
 
+	features, err := a.pubsubAdapter.PubSubFeatures(pubsubName)
+	if err != nil {
```

Comment:
There's one issue here where `PubSubFeatures` returns an error if the pub-sub component isn't found, and then it gets wrapped under the "ERR_PUBSUB_FEATURES_NOT_FOUND" which might be misleading to the user.

I would recommend to keep the existing error when pub/sub isn't found and separate that validation from the `PubSubFeatures` method.

Reply:
Agreed. Fixed.